src/families/cosmos/types.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import type { BigNumber } from "bignumber.js";
+import type {
+  TransactionCommon,
+  TransactionCommonRaw
+} from "../../types/transaction";
+
+export type CoreStatics = {};
+export type CoreAccountSpecifics = {};
+export type CoreOperationSpecifics = {};
+export type CoreCurrencySpecifics = {};
+
+export type CosmosDelegation = {|
+  validatorAddress: string,
+  amount: BigNumber,
+  pendingRewards: BigNumber,
+  isActive: boolean, // TBD is there more status case? if so, switch to status field with an enum string.
+|};
+
+export type CosmosRedelegation = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosUnbounding = {|
+  validatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosResources = {|
+  delegations: CosmosDelegation[],
+  redelegations: CosmosRedelegation[],
+  unbounding: CosmosUnbounding[],
+  delegatedBalance: BigNumber,
+  pendingRewardsBalance: BigNumber, 
+  unboundingBalance: BigNumber,
+  withdrawAddress: string,
+|};
+
+export type CosmosDelegationRaw = {|
+  validatorAddress: string,
+  amount: string,
+  pendingRewards: string,
+  isActive: boolean,
+|};
+
+export type CosmosRedelegationRaw = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosUnboundingRaw = {|
+  validatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosResourcesRaw = {|
+  delegations: CosmosDelegationRaw[],
+  redelegations: CosmosRedelegationRaw[],
+  unbounding: CosmosUnboundingRaw[],
+  delegatedBalance: string,
+  pendingRewardsBalance: string, 
+  unboundingBalance: string,
+  withdrawAddress: string,
+|};
+
+// NB this must be serializable (no Date, no BigNumber)
+export type CosmosValidatorItem = {|
+  validatorAddress: string,
+  name: string,
+  votingPower: number, // value from 0.0 to 1.0 (normalized percentage)
```

Comment:
Slight warning, we'll have to check how to compute this percentage, it might not be easy

Reply:
ok. we'll try and if it's a problem we'll remove that feature.

Reply:
We'll have to store either this or the votingPower in it's raw, bigNumber form anyway. It would be nicer to have the percentage, but I just don't know how to get it from the top of my head right now so it's more a "marker" to remember later to check this